### ISSUE TITLE
native support for autoscaling from zero

### ISSUE THREAD

**Issue by mweibel** [kubernetes-sigs/cluster-api-provider-azure#4771]
/kind feature

**Describe the solution you'd like**
I'd like to see native support for [cluster autoscaler scale from zero](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20210310-opt-in-autoscaling-from-zero.md). 

This would require adding the following `status` properties to AzureMachinePool, and AzureMachineTemplate:
```
    Capacity corev1.ResourceList `json:"capacity,omitempty"`
```

Which would need to be setup based on the instance type. 

**Environment:**

- cluster-api-provider-azure version: 1.14.2
- Kubernetes version: (use `kubectl version`): 1.28.5
- OS (e.g. from `/etc/os-release`): linux/windows

**Comment by mboersma** [comment]
@mweibel this sounds straightforward on the face of it. Is this something you intend to implement? We'd love to review the changes.

**Comment by mweibel** [comment]
yeah indeed, shouldn't be too hard to do. I'll pick that up when I find the time for sure. Probably will take a while though. If anyone else wants to, that would be great, too :)
